plugins/sinks/compass/sink.go

```diff
@@ -28,9 +28,10 @@ import (
 var summary string
 
 type Config struct {
-	Host    string            `mapstructure:"host" validate:"required"`
-	Headers map[string]string `mapstructure:"headers"`
-	Labels  map[string]string `mapstructure:"labels"`
+	Host              string            `mapstructure:"host" validate:"required"`
+	Headers           map[string]string `mapstructure:"headers"`
+	Labels            map[string]string `mapstructure:"labels"`
+	RemoveUnsetFields bool              `mapstructure:"remove_unset_fields" default:"false"`
```

Comment:
As discussed, Can you rename this field to denote its in asset/request something like - `RemoveUnsetFieldsInAsset`

Reply:
Changed it to `RemoveUnsetFieldsInData` as that looks more generic